src/utils.ts

```diff
@@ -48,9 +48,9 @@ function injectionCSSCodePlugin(cssToInject: string, styleId?: string): Plugin {
             if (id == cssInjectedByJsId) {
                 const cssCode = JSON.stringify(cssToInject.trim());
 
-                return `try{var elementStyle = document.createElement('style');${
-                    typeof styleId == 'string' && styleId.length > 0 ? `elementStyle.id = '${styleId}';` : ''
-                }elementStyle.appendChild(document.createTextNode(${cssCode}));document.head.appendChild(elementStyle);}catch(e){console.error('vite-plugin-css-injected-by-js', e);}`;
+                return `try{var elementStyle = typeof document !== 'undefined' && document.createElement('style');${
+                    typeof styleId == 'string' && styleId.length > 0 ? `elementStyle && elementStyle.id = '${styleId}';` : ''
+                }elementStyle && elementStyle.appendChild(document.createTextNode(${cssCode}));typeof document !== 'undefined' && document.head.appendChild(elementStyle);}catch(e){console.error('vite-plugin-css-injected-by-js', e);}`;
```

Comment:
I think since the global variable `document` is required for all statements and declarations inside the `try` block we can do a simple wrap it's whole code with `if (typeof document != 'undefined'){ ...the existing code in develop }`

Reply:
Yes! And then we can get rid of checking for `elementStyle` existence.
I'll update it and test it out a bit in multiple projects and let you know if it works fine.
Are there any tests needed to write or changelog file to update?

Reply:
Just tried it out and it works fine when SSR is not enabled. But with SSR I still get one `vite-plugin-css-injected-by-js ReferenceError: document is not defined` logged in the console (before this change it was shown 2 times).
From first glance this seems to me there is some other definition of `document.createElement("style")` in the plugin. I'll check it out

Reply:
For the SSR I will add the skip logic in another PR

Reply:
Ok 👍🏽 
Btw. This other ReferenceError was due to this plugin (without modification) used for icons project which ended up in the consumer project and triggering this error. So after applying changes there too, the issue is resolved. So can confirm from my side that everything works as before in browser environment, and those console logs are not longer visible with SSR. (Tried it in Next.js project)